File: Arrays_and_Strings/javascript/1.js

function isUniqueCharacters(word) {
    let map = new Set()

    for(let i=0; i< word.length; i++){
        char = word[i]
        if(!map.has(char)){
            map.add(char)
        } else {
            return false
        }
    }
    return true;
}

console.log(isUniqueCharacters("hello"))
console.log(isUniqueCharacters("hi"))
console.log(isUniqueCharacters("tale"))
console.log(isUniqueCharacters("straw"))
console.log(isUniqueCharacters("people"))

File: Arrays_and_Strings/python/1.py
def isUniqueCharacters(string: str):
    charSet = set()

    for char in string:
        if char in charSet:
            return False
        else:
            charSet.add(char)

    return True


print(isUniqueCharacters("hello"))
print(isUniqueCharacters("hi"))
print(isUniqueCharacters("tale"))
print(isUniqueCharacters("straw"))
print(isUniqueCharacters("people"))
